### DISCUSSION TITLE
Select next segment after "Split segment at cursor" command

### DISCUSSION THREAD

**Discussion by devtweakr** [mifi/lossless-cut#2135] [Ideas]
Add option in the settings, select next segment after cut.

**Comment by mifi** [comment]
what do you mean exactly by "select next segment after cut"?

**Comment by devtweakr** [comment]
Sorry, by cut , i meant "Split segment at cursor" command in keyboard shortcuts.

When you "Split segment at cursor", the segment before the split is still selected, so you have to manually select the segment after the split in order to work on it.

This is ok in some cases, so I would only add a toggle option in settings to enable to automatically select the segment after the split when you use "Split segment at cursor" command.

**Reply by mifi** [reply]
So what you mean is that you have a segment, let's call it X, and you press <kbd>B</kbd> to split it into to segments Y and Z, then segment Y gets selected. And you want segment Z to be selected instead? I think this can be achieved with an extra keyboard shortcut keypress (Jump to next segment). So basically when you want segment Z to be selected, then press these two keys: <kbd>B</kbd> then <kbd>↓</kbd>. In the cases where you want segment Y to be selected you only press <kbd>B</kbd>

**Reply by devtweakr** [reply]
Yes, that's exactly what I do now,

If there was a feature to write and assign scripts I would assign the B key to do:

Split segment at cursor
Jump to next segment

And this is where another genius idea pops up in my head (just kidding :)

Maybe add scripting feature for existing commands, so we can make our own automation scripts in javascript.
Then we could assign them to keys or add them to menus, right click commands, icons.
This could massively expand capabilities and add a ton of flexibility to UI.

This it could also ease your development of hard coding everything.

For example, this is the best feature of BLENDER with its console and scripting feature or REAPER DAW, they are basically a giant command library with a UI where everything can be scripted and automated. The users then created an entire ecosystem with thousands of scripts and add-ons.

**Reply by mifi** [reply]
thanks, i've added your idea here: #980 it would be cool indeed with a more dynamic/flexible command library to UI mapping
